aleph/index/entities.py

```diff
@@ -201,3 +206,32 @@ def delete_entity(entity_id, exclude=None, sync=False):
                   refresh=refresh_sync(sync))
         q = {'term': {'entities': entity_id}}
         query_delete(entities_read_index(), q, sync=sync)
+
+
+def delete_entities_by_mapping_id(stage, collection, mapping_id, sync=False):
+    """Delete entities loaded by a mapping"""
+    log.debug("Flushing entities for mapping %s", mapping_id)
+    aggregator = get_aggregator(collection)
+    index = entities_read_index()
+    query = {
+        'bool': {
+            'filter': [{
+                "term": {
+                    "mapping_id": mapping_id
+                }
+            }]
+        }
+    }
+    query = {
+        'query': query,
+        '_source': _source_spec([], []),
+        'size': MAX_PAGE
+    }
+    result = es.search(index=index, body=query)
+    # log.debug(result)
+    for doc in result.get('hits', {}).get('hits', []):
```

Comment:
I think this could easily use `aleph.index.collections:delete_entities` via an additional argument for `mapping_id`. That would still leave the aggregator, which we may want to drop entirely....